MonMaraicher/Features/Map/MapViewModel.swift

```diff
@@ -61,15 +67,24 @@ final class MapViewModel: ObservableObject {
 
     @MainActor private func loadFarmers() async {
         do {
-            let farmers = try await self.farmerService.loadFarmers()
+            progressLoadingOfFarmers = 0.0
+            farmersLoadingInProgress = true
+            guard let currentUserLocation else { return }
+            let farmers = try await self.farmerService.loadFarmers(latitude: currentUserLocation.coordinate.latitude, longitude: currentUserLocation.coordinate.longitude)
+            let totalMarkers = farmers.items.flatMap { $0.addresses }.count
+            var loadedMarkers = 0
             var allMarkers: [Marker] = []
             for farmer in farmers.items {
                 for address in farmer.addresses {
                     let marker = Marker(farmer: farmer, address: address)
                     allMarkers.append(marker)
+                    loadedMarkers += 1
+                    progressLoadingOfFarmers = Double(loadedMarkers) / Double(totalMarkers)
                 }
+                try await Task.sleep(for: .milliseconds(50))
```

Comment:
Oh! Why? Why making things longer? 😅

Reply:
You seem to simulate a loading right? Do you really want to do that? Do you think your user like to wait for the data?

Reply:
Yes good option imo